diff -urN node18/android-configure node18-patch/android-configure
--- node18/android-configure	2022-04-19 07:22:35.000000000 +0800
+++ node18-patch/android-configure	2022-04-23 12:23:24.459430100 +0800
@@ -8,14 +8,8 @@
 # modules with npm. Also, don't forget to set the arch in npm config using
 # 'npm config set arch=<arch>'
 
-if [ $# -ne 3 ]; then
-  echo "$0 should have 3 parameters: ndk_path, target_arch and sdk_version"
-  return 1
-fi
-
-NDK_PATH=$1
-ARCH="$2"
-ANDROID_SDK_VERSION=$3
+ARCH=arm64
+ANDROID_SDK_VERSION=28
 
 if [ $ANDROID_SDK_VERSION -lt 23 ]; then
   echo "$ANDROID_SDK_VERSION should equal or later than 23(Android 6.0)"
@@ -75,11 +69,25 @@
 GYP_DEFINES+=" host_os=$HOST_OS OS=android"
 export GYP_DEFINES
 
+PREFIX=~/dist/$ARCH
+
 if [ -f "configure" ]; then
     ./configure \
-        --dest-cpu=$DEST_CPU \
-        --dest-os=android \
-        --without-snapshot \
-        --openssl-no-asm \
-        --cross-compiling
+      --dest-cpu=$DEST_CPU \
+      --dest-os=android \
+      --cross-compiling \
+      --prefix=$PREFIX \
+      --v8-with-dchecks \
+      --verbose \
+      -C \
+      --without-node-snapshot \
+      --without-node-code-cache \
+      --without-npm \
+      --without-etw \
+      --without-report \
+      --without-dtrace \
+      --without-corepack \
+      --with-intl=none \
+      --openssl-no-asm \
+      --shared
 fi
diff -urN node18/deps/v8/src/trap-handler/trap-handler.h node18-patch/deps/v8/src/trap-handler/trap-handler.h
--- node18/deps/v8/src/trap-handler/trap-handler.h	2022-04-19 07:22:37.000000000 +0800
+++ node18-patch/deps/v8/src/trap-handler/trap-handler.h	2022-04-22 19:29:15.833234000 +0800
@@ -17,23 +17,7 @@
 namespace internal {
 namespace trap_handler {
 
-// X64 on Linux, Windows, MacOS, FreeBSD.
-#if V8_HOST_ARCH_X64 && V8_TARGET_ARCH_X64 &&                        \
-    ((V8_OS_LINUX && !V8_OS_ANDROID) || V8_OS_WIN || V8_OS_DARWIN || \
-     V8_OS_FREEBSD)
-#define V8_TRAP_HANDLER_SUPPORTED true
-// Arm64 (non-simulator) on Mac.
-#elif V8_TARGET_ARCH_ARM64 && V8_HOST_ARCH_ARM64 && V8_OS_DARWIN
-#define V8_TRAP_HANDLER_SUPPORTED true
-// Arm64 simulator on x64 on Linux, Mac, or Windows.
-#elif V8_TARGET_ARCH_ARM64 && V8_HOST_ARCH_X64 && \
-    (V8_OS_LINUX || V8_OS_DARWIN)
-#define V8_TRAP_HANDLER_VIA_SIMULATOR
-#define V8_TRAP_HANDLER_SUPPORTED true
-// Everything else is unsupported.
-#else
 #define V8_TRAP_HANDLER_SUPPORTED false
-#endif
 
 // Setup for shared library export.
 #if defined(BUILDING_V8_SHARED) && defined(V8_OS_WIN)
